Libraries/libutil/Sources/DefaultFilesystem.cpp

```diff
@@ -26,13 +26,34 @@
 #endif
 
 using libutil::DefaultFilesystem;
+using libutil::Filesystem;
 
 bool DefaultFilesystem::
 exists(std::string const &path) const
 {
     return ::access(path.c_str(), F_OK) == 0;
 }
 
+ext::optional<Filesystem::Type> DefaultFilesystem::
+type(std::string const &path) const
+{
+    struct stat st;
+    if (::lstat(path.c_str(), &st) < 0) {
+        return ext::nullopt;
+    }
+
+    if (S_ISREG(st.st_mode)) {
```

Comment:
Out of curiosity, would anything in xcbuild benefit from symbolic link knowledge being obscured?  Something like "I want to copy a file, I don't care if it's a real file or a symbolic link?"